lib/utilities/readToolbox.js

```diff
@@ -73,7 +73,7 @@ class ToolboxEntry {
    */
   getLinesByType(type) {
     return this.lines
-    .filter(line => line.type === type)
+    .filter(line => line.text !== undefined && line.type === type)
```

Comment:
These undefined cases were intentionally retained so that they can be used for various transformations on the Toolbox file (e.g. when I need to get a list of all the entries with empty `\src` fields so I can update those entries), and so that I can reconstitute the original Toolbox entry from the line information. This same `readToolbox.js` script gets used for all the work I'm doing with Arok on the Toolbox database, not just the Toolbox > JSON conversion script.

Can we push this filtering of empty lines to `convert/CW.js` instead? It'd be less DRY that way, I realize. I'm open to other solutions too.

(Side note: The `parseLine()` method should probably be updated so that the `text` property is an empty string rather than `undefined` in these cases.)

Reply:
Yup, these look good!